### PR TITLE
ballot contract and deploy script

### PR DESCRIPTION
**Pull request by AndyOooh**
## Description
New deploy command with args, split deploy scripts, add Ballot contract, minor configs.
deploy script in _package.json_ now takes` (contractArg, networkArg, ...constructorArgs)` where: 
- contractArg: name of contract in PascalCase, 
- networkArg: name of network in lowercase
- constructorArgs: list of arguments for the contracts constructor function.
- Example: `yarn deploy Ballot goerli banana apple orange`

## Motivation and Context
Set the project up to handle multiple contracts with one deploy command. 
Start weekend project: Ballot contract.

## How Has This Been Tested?
All previous passed tests pass. Deployment works for both HelloWorld and Ballot on testnet and goerli.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Style (style only changes)
- [x] Refactor (code that does not add new functionality nor fixes a bug)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   "main": "scripts/deploy.ts",
   "scripts": {
     "test": "npx hardhat test test/HelloWorld.ts",
-    "deploy": "npx hardhat run --network goerli scripts/deploy.ts",
+    "deploy": "yarn run ts-node --files scripts/deploy.ts",
```
Comment (evchip) [inline package.json:8]: We need to stick with one package manager. If you prefer yarn, I'm ok with using it. But in that case, please change all the scripts to use yarn.

File: scripts/deploy-ballot.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+if (!ALCHEMY_API_KEY) {
+  throw new Error('ALCHEMY_API_KEY missing');
+}
+
+if (!constructorArgs || constructorArgs.length === 0) {
+  throw new Error('You need to pass a valid constructor argument');
+}
+
+const PROPOSALS_BYTES32 = constructorArgs.map(p => ethers.utils.formatBytes32String(p));
```
Comment (evchip) [inline scripts/deploy-ballot.ts]: I recommend scoping all of this within the function. global scope variables are usually not desirable

File: scripts/deploy-ballot.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    for (let index = 0; index < constructorArgs.length; index++) {
+      const proposal = await contract.proposals(index);
+      proposals.push(proposal);
+    }
+    console.log(`Contract Ballot deployed to: ${contract.address} on chainId: ${
+      contract.deployTransaction.chainId
+    }
+    with proposals: ${proposals.map(p => ethers.utils.parseBytes32String(p.name))}`);
+  }
+
```
Comment (evchip) [inline scripts/deploy-ballot.ts]: Can just delete main and handle the promises with a try/catch

File: scripts/deploy-helloWorld.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    await helloWorld.deployed();
+
+    console.log(`🚀 ~ Hello World contract deployed to ${helloWorld.address}`);
+  }
+
+  main().catch(error => {
+    console.error(error);
+    process.exitCode = 1;
+  });
+};
```
Comment (evchip) [inline scripts/deploy-helloWorld.ts]: remove main and handle promises with try/catch

File: scripts/deploy.ts [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+  case 'HelloWorld':
+    deployHelloWorld();
+    break;
+  case 'Ballot':
+    deployBallot();
+    break;
+  default:
+    console.log('You need to pass a valid contract name as the first argument');
+    throw new Error('Invalid contract name');
 }
```
Comment (evchip) [inline scripts/deploy.ts]: Please wrap this in a function

**Review by evchip — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (AndyOooh) [reply package.json:8]: Ah yeah, my bad. Going with yarn then.

File: scripts/deploy-ballot.ts (hunk 2 above)
Comment (AndyOooh) [reply scripts/deploy-ballot.ts]: Done.

File: scripts/deploy-ballot.ts (hunk 3 above)
Comment (AndyOooh) [reply scripts/deploy-ballot.ts]: Done.

File: scripts/deploy-helloWorld.ts (hunk 4 above)
Comment (AndyOooh) [reply scripts/deploy-helloWorld.ts]: Done.

File: scripts/deploy.ts (hunk 5 above)
Comment (AndyOooh) [reply scripts/deploy.ts]: Done.

File: package.json (hunk 1 above)
Comment (AndyOooh) [reply package.json:8]: done.

**Review by evchip — APPROVED** [review]
LGTM

**Pull request merged**